plugins/plugin-testing-java/plugin-testing-junit/che-plugin-testing-junit-server/src/main/java/org/eclipse/che/plugin/testing/junit/server/junit4/TestRunner.java

```diff
@@ -0,0 +1,224 @@
+import org.eclipse.che.commons.lang.execution.JavaParameters;
+import org.eclipse.che.commons.lang.execution.ProcessHandler;
+import org.eclipse.che.plugin.java.testing.AbstractJavaTestRunner;
+import org.eclipse.che.plugin.java.testing.ClasspathUtil;
+import org.eclipse.che.plugin.java.testing.ProjectClasspathProvider;
+import org.eclipse.jdt.core.Flags;
+import org.eclipse.jdt.core.IAnnotation;
+import org.eclipse.jdt.core.ICompilationUnit;
+import org.eclipse.jdt.core.IImportDeclaration;
+import org.eclipse.jdt.core.IJavaElement;
+import org.eclipse.jdt.core.IJavaProject;
+import org.eclipse.jdt.core.IMethod;
+import org.eclipse.jdt.core.IType;
+import org.eclipse.jdt.core.JavaModelException;
+import org.eclipse.che.junit.junit4.CheJUnitCoreRunner;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.inject.Inject;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * JUnit implementation for the test runner service.
+ */
+public class TestRunner extends AbstractJavaTestRunner {
+    private static final Logger LOG = LoggerFactory.getLogger(TestRunner.class);
+
+    private static final String JUNIT_TEST_ANNOTATION = "org.junit.Test";
+    private static final String JUNIT_TEST_NAME       = "junit";
+    private static final String MAIN_CLASS_NAME       = "org.eclipse.che.junit.junit4.CheJUnitLauncher";
+
+    private ProjectClasspathProvider classpathProvider;
+
+    @Inject
+    public TestRunner(ProjectClasspathProvider classpathProvider) {
+        this.classpathProvider = classpathProvider;
+    }
+
+    @Override
+    public ProcessHandler execute(TestExecutionContext context) {
+        IJavaProject javaProject = getJavaProject(context.getProjectPath());
+        if (javaProject.exists()) {
+            return startTestProcess(javaProject, context);
+        }
+
+        return null;
+    }
+
+    @Override
+    public String getName() {
+        return JUNIT_TEST_NAME;
+    }
+
+    @Override
+    public boolean isTestMethod(IMethod method, ICompilationUnit compilationUnit) {
+        try {
+            int flags = method.getFlags();
+            if (method.isConstructor() ||
+                !Flags.isPublic(flags) ||
+                Flags.isAbstract(flags) ||
+                Flags.isStatic(flags) ||
+                !"V".equals(method.getReturnType())) { // 'V' is void signature
+                return false;
+            }
+
+            return isTest(method, compilationUnit);
+        } catch (JavaModelException ignored) {
+            return false;
+        }
+    }
+
+    private ProcessHandler startTestProcess(IJavaProject javaProject, TestExecutionContext context) {
+        JavaParameters parameters = new JavaParameters();
+        parameters.setJavaExecutable(JAVA_EXECUTABLE);
+        parameters.setMainClassName(MAIN_CLASS_NAME);
+
+        List<String> classPath = new ArrayList<>();
```

Comment:
Several testing frameworks or test runners (such as arquillian, etc ...) assume that the current working directory should be the root directory of the project.
See [this issue](https://issues.jboss.org/browse/CHE-214?focusedCommentId=13397893&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-13397893) as an example.
So it seems that the working directory should be explicitly set to the root directory of the project being tested.
This would also be consistent with the way tests are run inside maven afaik.
This has already been done for the TestNG part ( see [this discussion](https://github.com/eclipse/che/pull/5472#pullrequestreview-47609385) and [this line](https://github.com/eclipse/che/blob/67c144bd4c89316388c0ecf49b814dc1cd122cd9/plugins/plugin-testing-java/plugin-testing-testng/che-plugin-testing-testng-server/src/main/java/org/eclipse/che/plugin/testing/testng/server/TestNGRunner.java#L106) ).